### DISCUSSION TITLE
How to login using tsh in a Linux terminal

### DISCUSSION THREAD

**Discussion by PallaviPalisetti** [gravitational/teleport#14378] [Q&A] [answered]
Hi there, 
I set up oidc as a tsh login auth. I am authenticated from the browser whenever I log in to TSH using the OIDC authentication method. How does this functionality works if I use the VM Linux terminal and tsh login auth=oidc? (Linux server doesn't have any browser page). how to set it up so that it asks for my password rather than sending my request to the browser.

Thanks

**Comment by webvictim** [comment, marked as answer]
If your user is set up via OIDC, there is no way to log in with a password. You will need a browser to complete the OIDC flow.

There are a couple of workarounds:
1) Log in on your local machine with `tsh` and connect to the remote machine with SSH agent forwarding enabled (`tsh ssh -A` or `ssh -A`) - this will then let you connect to machines from there using your `tsh` identity. You can check your identity is available on the remote machine using `ssh-add -L`.

2) Forward a port from the remote machine to your local machine (`tsh ssh -L 54321:localhost:54321 user@hostname`) then use the `TELEPORT_LOGIN_BIND_ADDR` environment variable to set the port: `TELEPORT_LOGIN_BIND_ADDR=127.0.0.1:54321 tsh login --proxy=teleport.example.com:443`

When you load the URL that `tsh` pops up, it will complete the OIDC flow in your local browser and pass the result through to `tsh` on the remote machine.
